src/graphql/documents/projectEvents.graphql

```diff
@@ -0,0 +1,168 @@
+query ProjectEvents(
+  $where: ProjectEvent_filter
+  $orderBy: ProjectEvent_orderBy
+  $orderDirection: OrderDirection
+  $first: Int
+  $skip: Int
+) {
+  projectEvents(
+    where: $where
+    orderBy: $orderBy
+    orderDirection: $orderDirection
+    first: $first
+    skip: $skip
+  ) {
+    id
+    addToBalanceEvent {
+      id
+      timestamp
+      txHash
+      from
+      amount
+      note
+      terminal
+    }
+    burnEvent {
+      id
+      timestamp
+      txHash
+      from
+      holder
+      amount
+    }
+    configureEvent {
+      id
+      timestamp
+      txHash
+      from
+      ballot
+      dataSource
+      discountRate
+      duration
+      mintingAllowed
+      pausePay
+      redeemPaused
+      redemptionRate
+      reservedRate
+      weight
+      shouldHoldFees
+      terminalMigrationAllowed
+      controllerMigrationAllowed
+      setTerminalsAllowed
+      setControllerAllowed
+      useDataSourceForPay
+      useDataSourceForRedeem
+      memo
+      mustStartAtOrAfter
+      transfersPaused
+      burnPaused
+      distributionsPaused
+      useTotalOverflowForRedemptions
+    }
+    deployedERC20Event {
+      id
+      timestamp
+      txHash
+      from
+      symbol
+    }
+    deployETHERC20ProjectPayerEvent {
+      id
+      timestamp
+      txHash
+      from
+      address
+      memo
+    }
+    distributePayoutsEvent {
+      id
+      timestamp
+      txHash
+      from
+      beneficiary
+      beneficiaryDistributionAmount
+      distributedAmount
+      memo
+      terminal
+    }
+    distributeReservedTokensEvent {
+      id
+      timestamp
+      txHash
+      from
+      beneficiary
+      beneficiaryTokenCount
+      tokenCount
+    }
+    payEvent {
+      id
+      timestamp
+      txHash
+      from
+      amount
+      beneficiary
+      note
+      feeFromV2Project
+      terminal
+    }
+    printReservesEvent {
+      id
+      timestamp
+      txHash
+      from
+      beneficiary
+      beneficiaryTicketAmount
+      count
+    }
+    projectCreateEvent {
+      id
+      timestamp
+      txHash
+      from
+    }
+    redeemEvent {
+      id
+      timestamp
+      txHash
+      from
+      amount
+      beneficiary
+      returnAmount
+      terminal
+      metadata
+      memo
+    }
+    setFundAccessConstraintsEvent {
+      id
+      timestamp
+      txHash
+      from
+      distributionLimit
+      distributionLimitCurrency
+    }
+    tapEvent {
+      id
+      timestamp
+      txHash
+      from
+      beneficiary
+      beneficiaryTransferAmount
+      netTransferAmount
+    }
+    v1ConfigureEvent {
+      id
```

Comment:
there doesn't seem to be a way to `fragment` all these repeated property names, since these are all unique entities

subgraph schema doesn't support entities that `implement` an interface

https://www.apollographql.com/docs/react/data/fragments/

Reply:
@peripheralist could we fragment each event type though (e.g. `TapEvent`? I think that might work

Reply:
@tomquirk not sure I know what you mean, but if you're talking about creating a fragment on TapEvent that would apply to all these sub-events i.e. TapEvent, that's what i've tried and doesn't work. to do so, all sub-events would need to implement a similar interface, which the subgraph compiler doesn't let us do :c

Reply:
we could fragment TapEvent and apply it to all TapEvent references in our graphql files, but so far this is the only place we're referencing TapEvent